src/main/java/org/broadinstitute/hellbender/engine/Walker.java

```diff
@@ -0,0 +1,25 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+/**
+ * An abstract class to represent built-in walkers that inherit directly from {@link GATKTool}.
+ * Created by jonn on 6/28/18.
+ */
+public abstract class Walker extends GATKTool {
+
+    @Override
+    final protected ReferenceDataSource getReferenceDataSource() {
+        throw new GATKException("Should never access ReferenceDataSource in child classes of Walker.");
+    }
+
+    @Override
+    final protected ReadsDataSource getReadsDataSource() {
+        throw new GATKException("Should never access ReadsDataSource in child classes of Walker.");
+    }
+
+    @Override
```

Comment:
Add method-level javadoc comment stating that we are overriding this method to prevent walker tool implementations outside of the engine package from directly accessing the engine datasources, since walker tools should get their data via apply() instead.

Reply:
Fixed.